Configurable digdag metrics

yoyama opened the pull request:
This PR update digag-metrics as follows

- Digag-metrics is disabled as default
- Configurable by server config
- Each category is also configured enable/disable.

muga commented on digdag-server/src/main/java/io/digdag/server/metrics/DigdagMetricsConfig.java:
```diff
@@ -0,0 +1,62 @@
+
+public class DigdagMetricsConfig
+{
+    Logger logger = LoggerFactory.getLogger(DigdagMetricsConfig.class);
```
minor: should we make "private"?

muga replied:
for the field.

muga commented on digdag-server/src/main/java/io/digdag/server/metrics/DigdagMetricsConfig.java:
```diff
@@ -0,0 +1,62 @@
+public class DigdagMetricsConfig
+{
+    Logger logger = LoggerFactory.getLogger(DigdagMetricsConfig.class);
+    Map<String, DigdagMetricsPluginConfig> plugins = new HashMap<>();
```
ditto

muga commented:
Overall sounds good. Thank you for introducing that 👍 As we discussed in-person, let me summarize here, please confirm when you get a chance. 
- We'd better to revisit the appropriate place where "DigdagMetricsConfig" object is initialized now or later. In the current impl., that is in ServerConfig.java and it works.
- The current property name is "server.metrics.enable". But "metrics.enable" is more flexible in the future? for example, users may want to collect some metrics by local mode.
- The name of "DigdagMetricsPluginConfig" might make confusing because the "Plugin" comes from micrometer "plugin".
- In the future, will we allow customizing metrics collection by others (non "jmx" plugin) or not?

yoyama commented:
@muga Thanks for your reviews.

> We'd better to revisit the appropriate place where "DigdagMetricsConfig" object is initialized now or later. In the current impl., that is in ServerConfig.java and it works.

I removed DigdagMetricsConfig from ServerConfig.

>The current property name is "server.metrics.enable". But "metrics.enable" is more flexible in the future? for example, users may want to collect some metrics by local mode.

I fixed it.

>The name of "DigdagMetricsPluginConfig" might make confusing because the "Plugin" comes from micrometer "plugin".

Changed to MonitorSystemConfig. JMX, Datadog and others are called _monitor system_ in Micrometer site. So I named it.

>In the future, will we allow customizing metrics collection by others (non "jmx" plugin) or not?

Yes. 
Currently, DigdagEmbed and override some class is one of the way to do it.
We should support system plugin in future.
